support/oidc-discovery-provider/config.go

```diff
@@ -143,8 +155,13 @@ func ParseConfig(hclConfig string) (_ *Config, err error) {
 		c.LogLevel = defaultLogLevel
 	}
 
-	if c.Domain == "" {
-		return nil, errs.New("domain must be configured")
+	if c.Domain == "" && len(c.Domains) == 0 {
+		return nil, errs.New("at least one domain must be configured")
+	}
+
+	if c.Domain != "" {
+		c.Domains = append([]string{c.Domain}, c.Domains...)
+		c.Domains = unique(c.Domains)
```

Comment:
Not that it really matters, since they are just put into a set at a later point, but should we run `unique` either way? Nothing keeps folks from declaring the same domain more than once in the `Domains` configurable.

Reply:
It makes it at least more predictable for future code that uses the domains. Also covered this in a testcase.

Reply:
In that case, we should dedup `c.Domains` whether `c.Domain` was appended to it (i.e. move the dedup call outside, and after, the if block).